### ISSUE TITLE
func: uid() always returns data, even if node does not exist

### ISSUE THREAD

**Issue by aryzing** [dgraph-io/dgraph#2086]
Queries with `func: uid(<the_id>)` always return a node, even if the node does not exist. Is this the expected bahavior?

- What version of Dgraph are you using?
1.0.2

- Have you tried reproducing the issue with latest release?
yes

- Steps to reproduce the issue (command/config used to run Dgraph).
Issue query:

```
{
  random_node(func: uid(0x123abc456def)) {
    uid
  }
}
```

- Expected behaviour and actual result.
**Expected** The `random_node` query should be an empty array
**Actual** There's an object representing the queried id, but that node does not exist!
```
{
  "data": {
    "random_node": [
      {
        "uid": "0x123abc456def"
      }
    ]
  },
  "extensions": {
    "server_latency": {
      "parsing_ns": 13761,
      "processing_ns": 24076,
      "encoding_ns": 336092
    },
    "txn": {
      "start_ts": 50096,
      "lin_read": {}
    }
  }
}
```

**Comment by pawanrawal** [comment]
This is expected behaviour. Dgraph doesn't store nodes separately. We store all data for a `S P` combination together. A better way to check if a node exists will be to query for `_predicate_`. Though that would be empty again if you added and then deleted the data associated with a node.

```
{
  random_node(func: uid(0x123abc456def)) {
    _predicate_
  }
}
``` 

Curious to know how do you get the random `uid` if Dgraph didn't return it and what use case does checking for it serve?

**Comment by aryzing** [comment]
> what use case does checking for it serve?

Honestly, I don't know. I discovered this while playing with dgraph b/c I mistyped an id from a previous query. I was confused for quite some time until I realized the mistake due to there being an object with the erroneous id in the JSON response.

On my part, I made the assumption that querying for non-existant uids would either error out or return nothing, so I just wanted to confirm whether I had made the wrong assumption.

Thanks for the clarification!
